pkg/integrations/v2/app_o11y_receiver/exporters/receiver_metrics_test.go

```diff
@@ -0,0 +1,149 @@
+package exporters
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/grafana/agent/pkg/integrations/v2/app_o11y_receiver/models"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type metricAssertion struct {
+	name  string
+	value float64
+}
+
+func testcase(t *testing.T, payload models.Payload, assertions []metricAssertion) {
+	ctx := context.Background()
+
+	reg := prometheus.NewRegistry()
+
+	exporter := NewReceiverMetricsExporter(ReceiverMetricsExporterConfig{Reg: reg})
+
+	err := exporter.Export(ctx, payload)
+	assert.NoError(t, err)
+
+	metrics, err := reg.Gather()
+	assert.NoError(t, err)
+
+	for _, assertion := range assertions {
+		found := false
+		for _, metric := range metrics {
+			if *metric.Name == assertion.name {
+				found = true
+				assert.Len(t, metric.Metric, 1)
+				val := metric.Metric[0].Counter.Value
+				assert.Equal(t, assertion.value, *val)
+				break
+			}
+		}
+		if !found {
+			assert.Fail(t, fmt.Sprintf("metric [%s] not found", assertion.name))
+		}
+	}
+}
+
+func TestReceiverMetricsExport(t *testing.T) {
+	var payload models.Payload
+	payload.Logs = []models.Log{
+		{},
+		{},
+	}
+	payload.Measurements = []models.Measurement{
+		{},
+		{},
+		{},
+	}
+	payload.Exceptions = []models.Exception{
+		{},
+		{},
+		{},
+		{},
+	}
```

Comment:
This can be simplified pretty heavily:

```suggestion
        payload.Logs = make([]models.Log, 2)
        payload.Measurements = make([]models.Measurement, 3)
        payload.Exceptions = make([]models.Exception, 4)
```

ditto this suggestion for the other instances in this file